### PR TITLE
Feat/stacks 2.1 changelog

### PR DESCRIPTION
**Pull request by jcnelson**
Add 2.1 changelog

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,32 +5,106 @@ All notable changes to this project will be documented in this file.
...
-- Clarity2 improvements to traits (see #3251 for details):
+- There is a new `.pox-2` contract for implementing proof-of-transfer.  This PoX
+  contract enables re-stacking while the user's STX are locked, and incrementing
+the amount stacked on top of a locked batch of STX.
+- The Clarity function `stx-account` has been added, which returns the account's
+  locked and unlocked balances.
+- The Clarity functions `principal-destruct` and `principal-construct?`
+  functions have been added, which provide the means to convert between a
+`principal` instance and the `buff`s and `string-ascii`s that constitute it.
+- The Clarity function `get-burn-block-info?` has been extended to support
```
Comment (obycode) [inline CHANGELOG.md:27]: This a new function, not an extension of an existing one.

File: CHANGELOG.md [hunk 2]
```diff
@@ -5,32 +5,106 @@ All notable changes to this project will be documented in this file.
...
+- The Clarity function `replace-at?` has been added for replacing a single item
+  in a `list`, `string-ascii`, `string-utf8`, or `buff`.
+- The Clarity global variable `tx-sponsor?` has been added, which evaluates to
+  the sponsor of the transaction if the transaction is sponsored.
+- The Clarity global variable `chain-id` has been added, which evaluates to the
+  4-byte chain ID of this Stacks network.
+- The Clarity parser has been rewritten to be about 3x faster than the parser in
+  Stacks 2.05.x.x.x.
+- Clarity trait semantics have been refined and made more explicit, so as to
+  avoid certain corner cases where a trait reference might be downgraded to a
```
Comment (obycode) [inline CHANGELOG.md:52]: Maybe "principal" was supposed to be at the end of this sentence?

File: CHANGELOG.md [hunk 3]
```diff
@@ -5,32 +5,106 @@ All notable changes to this project will be documented in this file.
...
   * Trait values can be passed to compatible sub-trait types
   * Traits can be embedded in compound types, e.g. `(optional <my-trait>)`
   * Traits can be assigned to a let-variable
 - Fixes to unexpected behavior in traits
-  * A trait with duplicate function names is now an error (#3214)
-  * Aliased trait names do not interfere with local trait definitions (#3215)
+  * A trait with duplicate function names is now an error
+  * Aliased trait names do not interfere with local trait definitions
+`principal` in Clarity 1.
```
Comment (obycode) [inline CHANGELOG.md:59]: Oh, here's what was supposed to be the end of that sentence (see my previous comment).

**Review by kantai — APPROVED** [review]
🚀

File: CHANGELOG.md [hunk 4]
```diff
@@ -5,32 +5,106 @@ All notable changes to this project will be documented in this file.
...
+  can take effect within six (6) burnchain blocks in which they are mined,
+instead of one.
+- Transaction fees are debited from accounts _before_ the transaction is
+  processed.
+- All smart contract analysis errors are now treated as runtime errors, meaning
+  that smart contract transactions which don't pass analysis will still be mined
+(so miners get paid for partially validating them).
+- The default Clarity version is now 2.  Users can opt for version 1 by using
+  the new smart contract transaction wire format and explicitly setting version
+1.
```
Comment (obycode) [inline CHANGELOG.md:97]: This can be removed.

Comment (jcnelson) [reply CHANGELOG.md]: Fixed

File: CHANGELOG.md (hunk 2 above)
Comment (jcnelson) [reply CHANGELOG.md:52]: Fixed

File: CHANGELOG.md (hunk 1 above)
Comment (jcnelson) [reply CHANGELOG.md]: Fixed

**Review by obycode — APPROVED** [review]
lgtm. ship it!

**Review by pavitthrap — APPROVED** [review]

**Review by igorsyl — APPROVED** [review]

**Pull request merged**